### DISCUSSION TITLE
EULAs

### DISCUSSION THREAD

**Discussion by krosenfeld-IDM** [InstituteforDiseaseModeling/laser#24] [General]
From [Open Design Questions](https://github.com/InstituteforDiseaseModeling/laser/wiki/Open-Design-Questions):
> Jonathan has been working with the initial Epidemiologically Useless Light Agents as a complete separate cohort that is only modeled in a way that keeps track of population over time. Christopher has not yet optimized ignoring inactive (recovered or deceased) agents. See [here](https://github.com/InstituteforDiseaseModeling/laser/wiki/EULA) for more details.

@jonathanhhb original author

**Comment by krosenfeld-IDM** [comment]
@KevinMcCarthyAtIDM response:
>I believe the general concept of EULA cohorts can be incredibly useful for performance, as well as pushing users to really think about which agents drive the disease dynamics in their model. However, what a EULA really is can change depending on what we're modeling - the classic one is the "R" in an S(E)IR model. But there are other examples - people with access to clean water being uninteresting in a cholera model; non-sexually active agents being uninteresting in an STI model. So a question for software side about whether that implies something about generality vs. specificity of EULAs - e.g., should they be a "piece of the framework / reference design", or whether these are "performance tips / examples" applicable in specific model instantiations. Interested in thoughts. Note that we have not addressed, and may one day need to address, some intricacies that may arise. What happens if EULAs can become interesting one day? How to track things like mortality, planned events that target EULAs? How to ensure that the interesting agent <-> EULA transition is 1:1, so no agent gets double-counted...

**Comment by jonathanhhb** [comment]
My sense is that we've got some EULA and non-EULA options that we're comfortable with, some more work that we can do, but nothing here is impeding us from going forward.

I would like to see a write-up of @clorton's experience seeing high performance and no memory issues without EULAs (and large populations). I know I've had trouble with memory when experimenting with 1e7 literal agents. And it seems like working with 10x the number of agents would be 10x slower.

**Reply by krosenfeld-IDM** [reply]
My hope is that by doing some work now with numba but no EULA and then showing what numba + EULA can do we'll clearly see that improvement as well as the added complexity/requirements to enable it. So hopefully that will be a write-up / chart from that.
